### PR TITLE
Add `platformFilters` attribute

### PR DESCRIPTION
**Pull request by maxwellE**
Adds this attribute to PBXBuildFile and
PBXTargetDependency

### CODE REVIEW

File: Sources/XcodeProj/Objects/BuildPhase/PBXBuildFile.swift [hunk 1]
```diff
@@ -176,6 +183,9 @@ final class PBXBuildPhaseFile: PlistSerializable, Equatable {
         if let platformFilter = buildFile.platformFilter {
             dictionary["platformFilter"] = .string(.init(platformFilter))
         }
+        if let platformFilters = buildFile.platformFilters {
+            dictionary["platformFilters"] = .string(.init(platformFilters.joined(separator: ",")))
```
Comment (maxwellE) [inline Sources/XcodeProj/Objects/BuildPhase/PBXBuildFile.swift:187]: @brentleyjones I have no idea what separator should be used here

**Review by brentleyjones — APPROVED** [review]

**Review by kwridan — APPROVED** [review]
thanks for adding this.

A small suggestion for future contributions, it would be really helpful to capture where / when some of the attributes appear to aid testing and verifying the behaviours.

Luckily I was able to figure out where to look for this :)

In Xcode the filters option appears for targets and allows selecting all, single or multiple platforms

Filters

It appears selecting an individual platform filter results in the attribute `platformFilter` being used:

single-filter



```xml
3DFCA495EE0D3913DAD9804B /* Framework2.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = B7F0EC5C1D588EC72A1C5AEA /* Framework2.framework */; platformFilter = ios; };
```

vs selecting multiple platforms, results in the attribute `platformFilters` (plural) being used:

multiple-filters

```xml
3DFCA495EE0D3913DAD9804B /* Framework2.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = B7F0EC5C1D588EC72A1C5AEA /* Framework2.framework */; platformFilters = (ios, macos, tvos, ); };
```

From what I gather those are mutually exclusive. Xcode seems to prioritise `platformFilters` if both are present however upon making any changes would get rid on `platformFilter`. To maintain a backwards compatible API the approach taken here looks reasonable, but perhaps for a future major version we could consider consolidating the properties and internally have `XcodeProj` use the appropriate attribute based on single vs multiple filters.

**Comment by brentleyjones** [comment]
Just a counter-point, Xcode sometimes sets single element versions of the plural value:

```
platformFilters = (tvos, );
```

**Comment by maxwellE** [comment]
> thanks for adding this.
> 
> A small suggestion for future contributions, it would be really helpful to capture where / when some of the attributes appear to aid testing and verifying the behaviours.
> 
> Luckily I was able to figure out where to look for this :)
> 
> In Xcode the filters option appears for targets and allows selecting all, single or multiple platforms
> 
> Filters
> 
> It appears selecting an individual platform filter results in the attribute `platformFilter` being used:
> 
> single-filter
> 
> ```
> 3DFCA495EE0D3913DAD9804B /* Framework2.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = B7F0EC5C1D588EC72A1C5AEA /* Framework2.framework */; platformFilter = ios; };
> ```
> 
> vs selecting multiple platforms, results in the attribute `platformFilters` (plural) being used:
> 
> multiple-filters
> 
> ```
> 3DFCA495EE0D3913DAD9804B /* Framework2.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = B7F0EC5C1D588EC72A1C5AEA /* Framework2.framework */; platformFilters = (ios, macos, tvos, ); };
> ```
> 
> From what I gather those are mutually exclusive. Xcode seems to prioritise `platformFilters` if both are present however upon making any changes would get rid on `platformFilter`. To maintain a backwards compatible API the approach taken here looks reasonable, but perhaps for a future major version we could consider consolidating the properties and internally have `XcodeProj` use the appropriate attribute based on single vs multiple filters.

I will make sure to follow this workflow next time.

**Comment by kwridan** [comment]
> Just a counter-point, Xcode sometimes sets single element versions of the plural value:

fun times 😅

> I will make sure to follow this workflow next time.

Thank you

**Pull request merged**